macros/path_transformations/transform_paths.sql

```diff
@@ -0,0 +1,125 @@
+/* Macro to remove complexity from models paths_to_conversion / paths_to_non_conversion. */
+
+{% macro transform_paths(model_type) %}
+  {{ return(adapter.dispatch('transform_paths', 'snowplow_fractribution')(model_type)) }}
+{% endmacro %}
+
+{% macro default__transform_paths(model_type) %}
+
+  {% set allowed_path_transforms = ['exposure_path', 'first_path', 'frequency_path', 'remove_if_last_and_not_all', 'remove_if_not_all', 'unique_path'] %}
+
+  , path_transforms as (
+
+     select
+        customer_id,
+        {% if model_type == 'conversions' %}
+        conversion_tstamp,
+        revenue,
+        {% endif %}
+        {{ trim_long_path('path') }} as path,
+
+    {% if var('path_transforms').items()|length > 0 %}
+
+      {% for path_transform_name, _ in var('path_transforms').items()|reverse %}
+        {% if path_transform_name not in allowed_path_transforms %}
+          {%- do exceptions.raise_compiler_error("Snowplow Warning: the path transform - '"+path_transform_name+"' - is not supported. Please refer to the Snowplow docs on tagging. Please use one of the following: exposure_path, first_path, frequency_path, remove_if_last_and_not_all, remove_if_not_all, unique_path") %}
+        {% endif %}
+        {{target.schema}}.{{path_transform_name}}(
+      {% endfor %}
+
+      transformed_path
+
+      {% for _, transform_param in var('path_transforms').items()|reverse %}
+        {% if transform_param %}, '{{transform_param}}' {% endif %}
+        )
+      {% endfor %}
+
+      as transformed_path
+
+    {% else %}
+     transformed_path
+    {% endif %}
+
+  from arrays
+
+  )
+
+{% endmacro %}
+
+
+{% macro databricks__transform_paths(model_type) %}
+
+  {% set total_transformations = var('path_transforms').items()|length %}
+  {% set loop_count = namespace(value=1) %}
+
+  {% for path_transform_name, transform_param in var('path_transforms').items() %}
+
+    {%- if loop_count.value == 1 %}
```

Comment:
No (tested it), it will skip the path transform loop altogether and go do the trim long path one only: 
<img width="695" alt="image" src="https://user-images.githubusercontent.com/95634439/216138538-46e7e288-e5da-4a1d-a813-ee1c1c1623fa.png">